epoxy-adapter/src/main/java/com/airbnb/epoxy/EpoxyVisibilityTracker.kt

```diff
@@ -17,6 +17,11 @@ import java.util.HashMap
  * backed by an Epoxy controller. Once attached the events will be forwarded to the Epoxy model (or
  * to the Epoxy view when using annotations).
  *
+ * Note that support for visibility events on an [EpoxyModelGroup] is somewhat limited. Only model
+ * additions will receive visibility events. Models that are removed from the group will not receive
+ * events (e.g. [VisibilityState.INVISIBLE]) because the model group does not keep a reference,
+ * nor does it get notified of model removals.
```

Comment:
@eboudrant If the model group no longer has the model in it, it will not find that model's view on a view change, thus not re-compute visibility on it. If the model group _does_ still retain the model but the visibility changes then I think it should work just fine. I can try to add a test for that case.

Reply:
A test case would be nice.

Reply:
@eboudrant just added a couple of test cases for this. Hiding/showing models works as expected. The only outstanding problem is when the model is completely removed from the group. Looking into Eli's suggestion to see if we can do something there.